docs/extend/extend-cli/cli-developing-a-plugin.md

```diff
@@ -2,218 +2,323 @@
+
+2. Issue the `zowe plugins install` command to install Files Util Plug-in into Zowe CLI.
 
-Issue the following command. Replace the data set names with valid mainframe data set names on your system:
+    A success message displays if installed correctly.
 
-![Pipe Output](../../images/guides/CLI/pipeOutput.png)
+3. Issue the following command (the Files Util Plug-in command) replacing the data set names with valid mainframe data set names on your system:
 
-The raw diff output is displayed as a command response:
+    ```
+    zowe files-util diff data-sets "IBMUSER.REXX(TESTADD)" "IBMUSER.REXX(TESTSUB)"
+    ```
+    The raw diff output displays as a command response:
 
-![Raw Diff Output](../../images/guides/CLI/diffOutput.png)
+    ```
+    $ zowe files-util diff data-sets "IBMUSER.REXX(TESTADD)" "IBMUSER.REXX(TESTSUB)"
+    ===================================================================
+    --- IBMUSER.REXX(TESTADD)        Old
+    +++ IBMUSER.REXX(TESTSUB)        New
+    @@ -1,6 +1,6 @@
+    /* REXX */
+    -say add(3,2)
+    +say sub(3,2)
+    exit
+    -add:
+    +sub:
+    parse arg a,b
+    -return a + b
+    +return a - b
+    ```
+
+**To create a unique plug-in:** **[what can we advise here?]**
 
 ## Bringing together new tools!
-The advantage of Zowe CLI and of the CLI approach in mainframe development is that it allows for combining different developer tools for new and interesting uses.
 
-[diff2html](https://diff2html.xyz/) is a free tool to generate HTML side-by-side diffs to help see actual differences in diff output.
+You have created a simple CLI plug-in that provides plain text diffs of two data sets. But the nature of plain text can make it difficult to identify differences, depending on the changes.
+
+To help this, you can extend the plug-in to create a more visual output. For this tutorial, use [diff2html](https://diff2html.xyz/) to generate side-by-side diffs that make it easier to compare changes, as seen in the image below.
+
+| ![Side by Side Diff](../../images/guides/CLI/htmlDiff2.png) |
+|:--:|
+| <b>Diff to HTML by [rtfpessoa](https://github.com/rtfpessoa)</b>|
+
+1. Run `npm install -g diff2html-cli` to install `diff2html`. 
+
+2. To pipe your Zowe CLI plug-in output to `diff2html`, run the following command with your information:
+
+    ```
+    zowe files-util diff data-sets "IBMUSER.REXX(TESTADD)" "IBMUSER.REXX(TESTSUB)" | diff2html -i stdin
+    ```
 
-Install the `diff2html` CLI via `npm install -g diff2html-cli`. Then, pipe your Zowe CL plugin's output into `diff2html` to generate diff HTML and launch a web browser that contains the content in the screen shot at the [top of this file](#overview).
+    This launches a web browser that displays side-by-side diffs using HTML.
 
-- `zowe files-util diff data-sets "IBMUSER.work.jcl(iefbr14)" "IBMUSER.work.jcl(iefbr15)" | diff2html -i stdin`
+**To create a unique plug-in:** **[what can we advise here?]**
```

Comment:
This section will be very different depending on what command you have implemented. We could mention some examples of common ways to integrate with other tools, like piping output to another tool (in this case `diff2html`) or parsing JSON output in a script.